core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java

```diff
@@ -315,6 +319,49 @@ public boolean delete() {
     return new DataInputStream(new BufferedInputStream(stream));
   }
 
+  @Override public DataInputStream getDataInputStream(String path, FileFactory.FileType fileType,
```

Comment:
It already has a method in this class which chooses codec depends on the extension, Better use that instead of duplicating.

Reply:
Fine, I'll reduce the duplicate code.